app/models/job_io_wrapper.rb

```diff
@@ -0,0 +1,79 @@
+# Primarily for jobs like IngestJob to revivify an equivalent FileActor to one that existed on
+# the caller's side of an asynchronous Job invocation.  This involves providing slots
+# for the metadata that might travel w/ the actor's various supported types of @file.
+# For example, we cannot just do:
+#
+#   SomeJob.perform_later(arg1, arg2, File.new('/path/to/file'))
+#
+# Because we'll get:
+#
+#   ActiveJob::SerializationError: Unsupported argument type: File
+#
+# This also applies to Hydra::Derivatives::IoDecorator, Tempfile, etc., pretty much any IO.
+#
+# @note Along with user and file_set_id, path or uploaded_file are required.
+#  If both are provided: path is used preferentially for access IF it exists;
+#  however, the uploaded_file is used preferentially for default original_name and mime_type,
+#  because it already has that information.
+class JobIoWrapper < ApplicationRecord
+  belongs_to :user, optional: false
+  belongs_to :uploaded_file, optional: true, class_name: 'Hyrax::UploadedFile'
+  validates :uploaded_file, presence: true, if: proc { |x| x.path.blank? }
+  validates :file_set_id, presence: true
+
+  after_initialize :static_defaults
+  delegate :read, :size, to: :file
+
+  def original_name
+    super || extracted_original_name
+  end
+
+  def mime_type
+    super || extracted_mime_type
+  end
+
+  def file_set
+    FileSet.find(file_set_id)
+  end
+
+  def file_actor
+    Hyrax::Actors::FileActor.new(file_set, relation.to_sym, user)
```

Comment:
I considered that, but other jobs receiving `JobIoWrapper` objects are going to act directly via this `file_actor`.  The only reason it isn't currently directly invoked is the convenience method for `ingest_file`.  All _other_ actions (via the actor) would invoke `file_actor` or require their own convenience method.  I didn't want to preclude any methods in the actor from being used in jobs or force jobs to manually reinstantiate their own file actor.